state-sync/storage-service/server/src/lib.rs

```diff
@@ -209,41 +535,78 @@ fn refresh_cached_storage_summary<T: StorageReaderInterface>(
     Ok(())
 }
 
+/// Removes all expired data subscriptions
+fn remove_expired_data_subscriptions(
+    config: StorageServiceConfig,
+    data_subscriptions: Arc<Mutex<HashMap<AccountAddress, DataSubscriptionRequest>>>,
+) {
+    data_subscriptions.lock().retain(|_, data_subscription| {
+        !data_subscription.timed_out(config.max_subscription_timeout_ms)
```

Comment:
Can you clarify when would this timeout be useful - why can we not make it a no timeout subscription policy if  client just registers with the server and the server can keep sending any new updated data to the client?

Reply:
@JoshLind - IIUC SSv1 already does the "pure subscription" model right? how do we deal with the technical barriers for SSv1 (especially for network RPC request) in SSv1? In my view, the current model is a bit complex and also not so efficient because we are requiring clients to re-subscribe for data after receiving the update - which also increases the overhead on the server. If we can get as close to a "peer subscription" as possible, that will help scale the state sync. We can also take the subscription model a step further by coming up with "bit-torrent" like peer to peer data broadcast mechanism where the server don't need to put the same data on the wire several times - this is a bit complex though as we need to design this in mind keeping malicious clients.

Reply:
> IIUC SSv1 already does the "pure subscription" model right?

No 😄 SSv1 has this exact behaviour. Send a request to subscribe, wait for a notification, repeat.

> If we can get as close to a "peer subscription" as possible, that will help scale the state sync. 

Yeah, I agree. We're just not there yet 😄 At the very least, let's first match what SSv1 supports to help reduce the latency. Beyond that, we'll need to take a step back to understand where to go. In my honest opinion, we have way bigger fires everywhere else 🔥

Reply:
Thanks for clarifying - didn't know SSv1 has the same model.